Tools/paletteable sprite editor/js/marvinj-0.9.js

```diff
@@ -0,0 +1,2665 @@
+			}
+		}
+	}
+};
+
+MarvinImage.prototype.setColorToAlpha = function(alpha, color){
+	for(var y=0; y<this.height; y++){
+		for(var x=0; x<this.width; x++){
+			if((this.getIntColor(x,y) & 0x00FFFFFF) == (color & 0x00FFFFFF)){
+				this.setAlphaComponent(x,y,alpha);
+			}
+		}
+	}
+};
+
+MarvinImage.prototype.setAlphaToColor = function(color){
+	for(var y=0; y<this.height; y++){
+		for(var x=0; x<this.width; x++){
+			if(this.getAlphaComponent(x, y) == 0){
+				this.setIntColor(x, y, 0xFFFFFFFF);
+			}
+		}
+	}
+};
+
+MarvinImage.prototype.setIntColor2 = function(x,y, alpha, color){
+	var r = (color & 0x00FF0000) >> 16;
+	var g = (color & 0x0000FF00) >> 8;
+	var b = color & 0x000000FF;
+	this.setIntColor4(x,y,alpha,r,g,b);
+};
+
+MarvinImage.prototype.setIntColor3 = function(x,y, r, g, b){
+	this.setIntColor4(x,y,255,r,g,b);
+};
+
+MarvinImage.prototype.setIntColor4 = function(x,y, alpha, r, g, b){
+	var start = ((y*this.getWidth())+x)*4;
+	this.imageData.data[start] = r;
+	this.imageData.data[start+1] = g;
+	this.imageData.data[start+2] = b;
+	this.imageData.data[start+3] = alpha;
+};
+
+MarvinImage.prototype.getWidth = function(){
+	return this.width;
+};
+
+MarvinImage.prototype.getHeight = function(){
+	return this.height;
+};
+
+MarvinImage.prototype.isValidPosition = function(x, y){
+	if(x >= 0 && x < this.width && y >= 0 && y < this.height){
+		return true;
+	}
+	return false;
+};
+
+MarvinImage.prototype.draw = function(canvas, x, y, alphaCombination){
+	if(x == null){x=0;}
+	if(y == null){y=0;}
+	canvas.getContext("2d").putImageData(this.imageData, x,y);/*
+	if(alphaCombination == null || !alphaCombination){
+		canvas.getContext("2d").putImageData(this.imageData, x,y);
+	} else{
+		this.imageData = this.ctx.getImageData(0, 0, width, height);
+		var c = document.createElement('canvas');
+		c.width = this.width;
+		c.height = this.height;
+		c.getContext('2d').putImageData(this.imageData,x,y); 
+		var img = new Image();
+		img.src = c.toDataURL();
+		canvas.getContext("2d").drawImage(img, x, y);
+	}*/
+};
+
+MarvinImage.prototype.toBlob = function(){
+	this.update();
+	return MarvinImage.dataURItoBlob(this.canvas.toDataURL("image/png"));
+};
+
+MarvinImage.dataURItoBlob = function(dataURI) {
+    // convert base64/URLEncoded data component to raw binary data held in a string
+    var byteString;
+    if (dataURI.split(',')[0].indexOf('base64') >= 0)
+        byteString = atob(dataURI.split(',')[1]);
+    else
+        byteString = unescape(dataURI.split(',')[1]);
+
+    // separate out the mime component
+    var mimeString = dataURI.split(',')[0].split(':')[1].split(';')[0];
+
+    // write the bytes of the string to a typed array
+    var ia = new Uint8Array(byteString.length);
+    for (var i = 0; i < byteString.length; i++) {
+        ia[i] = byteString.charCodeAt(i);
+    }
+
+    return new Blob([ia], {type:mimeString});
+}
+
+
+	
+	function MarvinImageMask (w, h){
+		this.width = w;
+		this.height = h;
+		
+		if(w != 0 && h != 0){
```

Comment:
Codacy found an issue: [Use ===/!== to compare with true/false or Numbers](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6395467)